### PR TITLE
Adding docstrings and typing

### PR DESCRIPTION
**Pull request by schafrn**
What:
Added in docstrings following Google style (https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) , attempting to make things clearer and easier to use.
Added typing to where it was missing.
Ran auto-formatting on the files.

Why:
The code was unclear and hard to get up-to-speed on or understand after not touching it for a while. Adding in the docstrings gives a clearer guide and better IDE integrations. Keeping to a defined style helped keep them consistent as well.
Typing just to fix up IDE hints.

### CODE REVIEW

File: s3parq/fetch_parq.py [hunk 1]
```diff
@@ -370,13 +470,26 @@ def construct_paths(matched_parts, previous_fil_keys: List[str]) -> None:
     return filter_keys[0]
 
 
-def _get_filtered_data(bucket: str, paths: list, partition_metadata, parallel=True) -> pd.DataFrame:
-    ''' Pull all filtered parquets down and return a dataframe.
-    '''
+def _get_filtered_data(bucket: str, paths: List[str], partition_metadata: dict, parallel: bool =True) -> pd.DataFrame:
```
Comment (RyanAdalbert) [inline s3parq/fetch_parq.py]: So so close to perfection `parallel: bool =True)`

**Review by RyanAdalbert — APPROVED** [review]
Outstanding job cleaning up the code and the docstrings! Awesome work Rayne

File: s3parq/publish_parq.py [hunk 2]
```diff
@@ -232,59 +388,85 @@ def log_size_estimate(num_bytes):
...
         redshift_params = validate_redshift_params(redshift_params)
         logger.debug("Redshift parameters valid. Opening Session helper.")
         session_helper = SessionHelper(
-            region = redshift_params['region'],
-            cluster_id = redshift_params['cluster_id'],
-            host = redshift_params['host'],
-            port = redshift_params['port'],
-            db_name = redshift_params['db_name'],
-            ec2_user = redshift_params['ec2_user']
+            region=redshift_params['region'],
```
Comment (jlew25) [inline s3parq/publish_parq.py:437]: you don't like my spaces? :(

File: s3parq/fetch_parq.py [hunk 3]
```diff
@@ -177,14 +211,27 @@ def fetch(bucket: str, key: str, filters: List[type(Filter)] = {}, parallel: boo
...
+        reverse (bool, Optional): Determines if the operation should be inversed,
+            if True it will look for the values in comparison that are not 
+            in the input (basically backwards). Defaults to False
+        parallel (bool, Optional):
+            Determines if multiprocessing should be used, defaults to True.
+            Whether this is more or less efficient is dataset dependent,
+            smaller datasets run much quicker without parallel
+
+    Returns:
+        An dataframe of all values in the input but not the comparison,
```
Comment (jlew25) [inline s3parq/fetch_parq.py]: "An" -> "A"

**Review by jlew25 — COMMENTED** [review]
Looks good to me!  There was just one grammar thing I pointed out 👍

File: s3parq/publish_parq.py (hunk 2 above)
Comment (schafrn) [reply s3parq/publish_parq.py:437]: Full disclosure, I do prefer the spaces but pep8 does not :( (I auto-pep8-ed the files I touched)

Comment (jlew25) [reply s3parq/publish_parq.py:437]: I see I see

**Pull request merged**